core/src/main/java/org/web3j/protocol/http/HttpService.java

```diff
@@ -185,9 +196,13 @@ protected void processHeaders(Headers headers) {
     private InputStream buildInputStream(ResponseBody responseBody) throws IOException {
         if (includeRawResponse) {
             return new ByteArrayInputStream(responseBody.bytes());
-        } else {
-            return responseBody.byteStream();
         }
+
+        BufferedSource source = responseBody.source();
+
+        source.request(Long.MAX_VALUE); // Buffer the body
```

Comment:
This will force the response to be loaded, which shouldn't be the case if includeRawResponse is false

Reply:
then a solution will be to clone the responseBody.byteStream()

or refactor HttpsService (also for batch requests) and process them inside the try block